web_hierarchy/static/src/js/web_hierarchy_view.js

```diff
@@ -0,0 +1,95 @@
+            var size = (!size) ? "medium": size;  // default value
+            var parent_name = this.options.action.display_name;
+            google.charts.load('current', {packages: ["orgchart"]});
+            google.charts.setOnLoadCallback(drawChart);
+
+            function drawChart() {
+                var vals = [];
+                var data = new google.visualization.DataTable();
+                data.addColumn('string', 'ID')
+                data.addColumn('string', 'Name');
+                data.addColumn('string', 'ToolTip');
+                self.dataset.read_slice(self.fields_view.fields, {}).then(
+                    function (results) {
+                        var root = parent_name || "Root"
+                        vals.push([{v: root, f: root}, '', 'root element']);
+                        for (var i = 0; i < results.length; i++) {
+                            if (results[i].parent_id) {
+                                if (results[i].parent_id[1].search("/")) {
+                                    var x = results[i].parent_id[1].split("/");
+                                    var parent = x[x.length - 1];
+                                    vals.push([results[i].name, parent.trim(),
+                                        results[i].name]);
+                                } else {
+                                    vals.push([results[i].name,
+                                        results[i].parent_id[1].trim(),
+                                        results[i].name]);
+                                }
+                            } else {
+                                vals.push([results[i].name, root, root]);
+                            }
+                        }
+                        data.addRows(vals);
+                        // Create the chart.
+                        var chart = new google.visualization.OrgChart(
+                            document.getElementById('chart_div'));
+                        // Draw the chart, setting the allowHtml option to
+                        // true for the tooltips.
+                        chart.draw(data, {allowHtml: true, size: size});
+                    });
+            }
+        },
+        do_show: function () {
+            this.do_push_state({});
+            this.render_chart();
+            this._super.apply(this, arguments);
+        },
+        do_add_record: function () {
+            this.dataset.index = null;
+            _.delay(_.bind(function () {
+                this.do_switch_view('form');
+            }, this));
+        },
+        render_buttons: function ($node) {
+            var self = this;
+            if (!this.$buttons) {
+                this.$buttons = $(qweb.render("HierarchyView.buttons",
+                    {'widget': this}));
+                this.$buttons.on('click', '.o_org_button_add',
+                    this.proxy('do_add_record'));
+                this.$buttons.on('click', '.o_select_btn',
```

Comment:
This does not work very well. The rendering of the chart should happen on when the value of the `select` changes, not when clicked, check this out https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event